server/src/sim/makeccocs.py

```diff
@@ -12,132 +12,117 @@
+    if popadj: totalcost = totalcost/targetpopsize if len(totalcost)>1 else totalcost/mean(targetpopsize)
 
     # Get upper limit of x axis for plotting
     xupperlim = max([x if ~isnan(x) else 0.0 for x in totalcost])*1.5
-#    if (ccplot and ccplot[0]): xupperlim = max(xupperlim, ccplot[0]) if not (len(ccplot)==3 and ccplot[2]) else max(xupperlim, ccplot[0]/targetpopsize[-1]) if len(totalcost)>1 else max(xupperlim, ccplot[0]/mean(targetpopsize)) 
-    if (ccplot and ccplot[0]): xupperlim = ccplot[0] #if not (len(ccplot)==3 and ccplot[2]) else max(xupperlim, ccplot[0]/targetpopsize[-1]) if len(totalcost)>1 else max(xupperlim, ccplot[0]/mean(targetpopsize)) 
-        
-    # Populate output structure with scatter data 
+    if (ccparams and 'xupperlim' in ccparams and ~isnan(ccparams['xupperlim'])): xupperlim = ccparams['xupperlim']
+
+    # Populate output structure with scatter data
     totalcost, coverage = getscatterdata(totalcost, coverage)
     plotdata['xscatterdata'] = totalcost
     plotdata['yscatterdata'] = coverage
 
-    # Are there parameters (either given by the user or previously stored)?
-    if (ccparams or D['programs'][progname]['ccparams']):
-        if not ccparams:
-            ccparams = D['programs'][progname]['ccparams']
-        coverage, targetpopsize, coveragelabel, convertedccparams, ccplottingparams = getcoverage(D, ccparams, popadj=popadj, artelig=default_artelig, progname=progname)
-        
-        # Create curve
+    # Are the curve-making parameters there?
+    reqccparams = [ccparams.get('coveragelower'), ccparams.get('coverageupper'), ccparams.get('funding'), ccparams.get('saturation')]
+    if all(reqccparams) and all(~isnan(reqccparams)):
+
+        convertedccparams = convertparams(D=D, ccparams=ccparams)
+
+        # X data
         xvalscc = linspace(0,xupperlim,nxpts) # take nxpts points between 0 and user-specified max
-        if isinstance(ccparams[4], float):
-            yvalsccl = cceqn(xvalscc, ccplottingparams[0])
-            yvalsccm = cceqn(xvalscc, ccplottingparams[1])
-            yvalsccu = cceqn(xvalscc, ccplottingparams[2])
+        xvalsccpop = linspace(0,xupperlim*targetpopsize[-1],nxpts) if popadj else xvalscc
+
+        # Y data
+        if 'scaleup' in ccparams:
+            yvalsccl = cceqn(xvalsccpop, convertedccparams[0])
+            yvalsccm = cceqn(xvalsccpop, convertedccparams[1])
+            yvalsccu = cceqn(xvalsccpop, convertedccparams[2])
             yvalscc = [yvalsccl, yvalsccm, yvalsccu]
         else:
-            yvalsccl = cc2eqn(xvalscc, ccplottingparams[0])
-            yvalsccm = cc2eqn(xvalscc, ccplottingparams[1])
-            yvalsccu = cc2eqn(xvalscc, ccplottingparams[2])
+            yvalsccl = cc2eqn(xvalsccpop, convertedccparams[0])
+            yvalsccm = cc2eqn(xvalsccpop, convertedccparams[1])
+            yvalsccu = cc2eqn(xvalsccpop, convertedccparams[2])
             yvalscc = [yvalsccl, yvalsccm, yvalsccu]
```

Comment:
On line 130 of makeccocs.py the variable yvalscc is referenced. Since it's initialised in an if depending on reqccparams with one program it was never initialised & it got this error:

    UnboundLocalError: local variable 'yvalscc' referenced before assignment

@robynstuart what should `yvalscc` be in this case?